src/gui/qgsfieldmappingmodel.cpp

```diff
@@ -342,7 +342,7 @@ void QgsFieldMappingModel::setDestinationFields( const QgsFields &destinationFie
   mMapping.clear();
   // Prepare the model data
   QStringList usedFields;
-  for ( const auto &df : qgis::as_const( destinationFields ) )
```

Comment:
Side question regarding this:

One thing that makes me not like range-based for-loops as much as the good old Q_FOREACH is it's notorious dependency on const. And I was wondering if we should simply just always add `qgis::as_const` even when redundant because it makes code reviewing so much easier.